core/src/ncipher/aes_gcm.c

```diff
@@ -26,6 +26,13 @@ uint8_t aes_gcm_buffer[1000];
 Result aes_gcm_encrypt(M_KeyID keyId, uint8_t *plaintext, size_t plaintext_len,
                        uint8_t *ciphertext, size_t ciphertext_len,
                        size_t *bytes_written) {
+
+  // NULL pointer checks. Note that plaintext can be NULL.
+  if (!ciphertext || !bytes_written) {
+    ERROR("%s: ciphertext and bytes_written must not be NULL", __func__);
```

Comment:
`ERROR` is a macro which already includes the filename + line number (see https://github.com/square/subzero/blob/42044320c0909515b48706b10fe824bb99b51f27/core/include/log.h#L40)

Reply:
Hmm, it seems complimentary to add a function name in this case, like https://github.com/square/subzero/blob/e91e40a84634dc4bccc09d593c40dbd500d2ac8a/core/src/ncipher/aes_gcm.c#L39. Note that the ERROR macro includes the filename and line number, but not function name. What do you think, @alokmenghrajani ?

Reply:
I think it was a mistake on my part to manually include the function name. If the function name adds value, we should just update the macro and stop doing it manually?

Reply:
It seems to me that function name can come in handy for debugging - it provides additional information after all. The ERROR macro is currently used in many places in the code base, where occasionally hard coded function names are included in the error strings. I'm leaning towards keeping the macro AS IS in this PR, and in a separate PR update it to include the function name, and do some cleanups accordingly. What do you think?

Reply:
sgtm